clip-recap-pipeline/src/utils/validation.ts

```
@@ -0,0 +1,226 @@
+
+// Sanitize string inputs to remove dangerous characters
+export function sanitizeString(input: string): string {
+  return input
+    // Remove HTML tags
+    .replace(/<[^>]*>/g, '')
+    // Remove script tags and their content
+    .replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '')
+    // Remove dangerous characters that could be used for XSS
+    .replace(/[<>\"'&]/g, '')
+    // Remove control characters
+    .replace(/[\x00-\x1F\x7F]/g, '')
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)